Common/Tests/Utilities/AssertUtil.cs

```diff
@@ -68,29 +68,20 @@ public static void Throws<TExpected>(Action throwingAction, string description)
             Assert.Inconclusive("Missing Dependency: {0}", dependency);
         }
 
-        public static void ArrayEquals(IList expected, IList actual)
-        {
-            if (expected == null)
-            {
+        public static void ArrayEquals<T>(IEnumerable<T> expected, IEnumerable<T> actual) {
+            if (expected == null) {
                 throw new ArgumentNullException("expected");
             }
-            if (actual == null)
-            {
-                Assert.Fail("AssertUtils.ArrayEquals failure. Actual collection is null.");
+            if (actual == null) {
+                Assert.Fail("Actual collection is null.");
             }
 
-            if (expected.Count != actual.Count)
-            {
-                Assert.Fail("AssertUtils.ArrayEquals failure. Expected collection with length {0} but got collection with length {1}",
-                    expected.Count, actual.Count);
-            }
-            for (int i = 0; i < expected.Count; i++)
-            {
-                if (!expected[i].Equals(actual[i]))
-                {
-                    Assert.Fail("AssertUtils.ArrayEquals failure. Expected value {0} at position {1} but got value {2}",
-                        expected[i], i, actual[i]);
-                }
+            Assert.AreEqual(expected.Count(), actual.Count());
+
+            var expectedIt = expected.GetEnumerator();
+            var actualIt = actual.GetEnumerator();
+            for (var i = 0; expectedIt.MoveNext() && actualIt.MoveNext(); ++i) {
+                Assert.AreEqual(expectedIt.Current, actualIt.Current, string.Format("Assetion failed at index {0}", i));
             }
```

Comment:
IEnumerable does not support random access. I can revert this change because it's not strictly used anymore, but the old definition is more restrictive than needed.